docs/admin/external-auth.md

```diff
@@ -27,62 +33,62 @@ The `CODER_EXTERNAL_AUTH_0_ID` environment variable is used for internal
 reference. Therefore, it can be set arbitrarily (e.g., `primary-github` for your
 GitHub provider).
 
-## GitHub
+Add the following code to any template to add a button to the workspace setup page which will allow you to authenticate with your provider:
 
-> If you don't require fine-grained access control, it's easier to configure a
-> GitHub OAuth app!
+```tf
+data "coder_external_auth" "<github|gitlab|azure-devops|bitbucket-cloud|bitbucket-server|etc>" {
+    id = "<USER_DEFINED_ID>"
+}
 
-1. [Create a GitHub App](https://docs.github.com/en/apps/creating-github-apps/registering-a-github-app/registering-a-github-app)
```

Comment:
this should also be prepended with a `#` otherwise if someone directly copies it might fail. Also, the link won't render in code block. or Do they?